woocommerce/woocommerce-ios
# Google Ads: Optimize dashboard card and handle navigation

Closes: #13230 


## Description

This PR completes the Google ads dashboard card features:
- Handled the navigation to campaign creation and dashboard. This includes updating the coordinator for a more generic navigation logic.
- Updated fetching campaign stats to support filtering by campaign ID.
- Updated the data loading on the view model for easier testing.
- Added unit tests.

## Testing steps

Note: Test campaigns don't get impressions or clicks, so the results are always 0. To test non-0 cases, please feel free to stub the response using proxyman for `GET wc/gla/ads/reports/programs` endpoint.

- Pre-requisite: Install, activate, and set up the Google Ads plugin to a test site following pe5sF9-2Qo-p2. Ensure that you're using the plugin version 2.7.5 or above.
- Build the app and log in to your store.
- Enable Google Ads card if you haven't already.
- Create a Google Ads campaign if you haven't already and reload the dashboard.
- Confirm that the last campaign details are displayed correctly.
- Tap on the campaign details, confirm that Google Ads dashboard page is presented.
- Tap on the Create campaign button, confirm that Google Ads campaign creation page is presented.


## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/9670ad55-f9d7-4317-ae15-0f0ce0a51529" width=320 />

---
